source/lib-intro.tex

```diff
@@ -820,6 +859,47 @@
 A \defnx{static \ntmbs{}}{NTMBS@\ntmbs{}!static}
 is an \ntmbs{} with static storage duration.
 
+\rSec4[customization.point.object]{Customization Point Object types}
+
+\pnum
+A \term{customization point object} is a function object\iref{function.objects}
+with a literal class type that interacts with user-defined types while enforcing
+semantic requirements on that interaction.
+
+\pnum
+The type of a customization point object shall satisfy
+\tcode{Semiregular}\iref{concepts.lib.object.semiregular}.
+
+\pnum
+All instances of a specific customization point object type shall
+be equal\iref{concepts.lib.general.equality}.
+
+\pnum
+The type \tcode{T} of a customization point object shall satisfy
+\tcode{Invocable<const T\&, Args...>}\iref{concepts.lib.callable.invocable} when
+the types in \tcode{Args...} meet the requirements specified in that
+customization point object's definition. When the types of \tcode{Args...} do not
+meet the customization point object's requirements, \tcode{T} shall not have a
+function call operator that participates in overload resolution.
+
+\pnum
+Each customization point object type constrains its return type to satisfy a
+particular concept.
+
+\pnum
+\begin{note}
+Many of the customization point objects in the library evaluate function call
+expressions with an unqualified name which results in a call to a user-defined
+function found by argument dependent name lookup\iref{basic.lookup.argdep}. To
+preclude such an expression resulting in a call to unconstrained functions with
+the same name in namespace \tcode{std}, customization point objects specify that
+lookup for these expressions is performed in a context that includes deleted
+overloads matching the signatures of overloads defined in namespace \tcode{std}.
+When the deleted overloads are viable, user-defined overloads must be more
+specialized\iref{temp.func.order} or more constrained\iref{temp.constr.order} to
+be used by a customization point object.
```

Comment:
"more constrained" is a special case of "more specialized".

Reply:
[temp.constr.order]/3 defines "more constrained" as an orthogonal notion to "more specialized." The two notions interact in overload resolution where "more constrained" is used to discriminate between overloads neither of which is more specialized, but "more constrained" is not otherwise a special case.